### ISSUE TITLE
x/website/_content/ref: document how licenses are added to modules that are nested in a repository

### ISSUE THREAD

**Issue by mdempsky** [golang/go#44758]
https://pkg.go.dev/github.com/mit-plv/fiat-crypto/fiat-go

The fiat-crypto project is triple licensed under MIT OR Apache-2.0 OR BSD-1-Clause (https://github.com/mit-plv/fiat-crypto/blob/master/COPYRIGHT), but pkg.go.dev isn't recognizing this. It's instead saying the license isn't known.

Incidentally, it seems like maybe https://pkg.go.dev/license-policy should be updated to support BSD-1-Clause?

Related: #40171

**Comment by mdempsky** [comment]
So https://pkg.go.dev/github.com/mit-plv/fiat-crypto/fiat-go@v0.0.0-20210303142032-cc0899379641/32/curve25519 now correctly shows the license, but https://pkg.go.dev/github.com/mit-plv/fiat-crypto/fiat-go@v0.0.0-20210303132348-f524ab755084/32/curve25519 still doesn't. Presumably this is because of https://github.com/mit-plv/fiat-crypto/commit/d12f145229de4ed4b664b8747c70e280a8442425.

For repos where there's no go.mod at the top level, is it expected that the copyright files have to be explicitly duplicated into the go.mod-containing subdirectory? That seems surprising to me.

**Comment by julieqiu** [comment]
> For repos where there's no go.mod at the top level, is it expected that the copyright files have to be explicitly duplicated into the go.mod-containing subdirectory? 

Yes, pkg.go.dev fetches modules from proxy.golang.org, and those license files are not in https://proxy.golang.org/github.com/mit-plv/fiat-crypto/fiat-go/@v/v0.0.0-20210303132348-f524ab755084.zip.

> Incidentally, it seems like maybe https://pkg.go.dev/license-policy should be updated to support BSD-1-Clause?

Yup, BSD-1-Clause should listed on https://pkg.go.dev/license-policy. Thanks for catching.

**Comment by mdempsky** [comment]
Why does https://proxy.golang.org/golang.org/x/tools/gopls/@v/v0.6.6.zip contain ./golang.org/x/tools/gopls@v0.6.6/LICENSE when that file does not appear in https://github.com/golang/tools/tree/master/gopls?

**Comment by julieqiu** [comment]
My understanding is that files named LICENSE in the top level module (golang.org/x/tools in this case) will be added its nested modules (golang.org/x/tool/gopls). Related: https://github.com/golang/go/issues/43817


/cc @katiehockman @heschik @hyangah

**Comment by mdempsky** [comment]
Reading through that issue, I understand the difficulties now. Thanks.

Is this behavior documented somewhere? The inconsistency here seems pretty surprising to me.

**Comment by julieqiu** [comment]
Not that I know of. 

/cc @stevetraut @bcmills @jayconrod @matloob who might be working on documentation for this.

**Comment by jayconrod** [comment]
This probably should be documented, at least in https://golang.org/ref/mod. Anywhere else?

**Comment by julieqiu** [comment]
It might also be useful to add this to the FAQ of https://proxy.golang.org.

**Comment by mdempsky** [comment]
FWIW, I notice that https://pkg.go.dev/github.com/mit-plv/fiat-crypto/fiat-go?tab=licenses lists MIT and Apache 2, but not BSD-1. I guess that's because it's in a LICENSE-BSD file, which pkgsite doesn't look for.

Is it possible to have that added too? Or maybe pkgsite should check any file in the root directory that matches "\bLICEN[CS]E\b"?

**Comment by hyangah** [comment]
The go command's handling of LICENSE is documented in https://go.dev/ref/mod#vcs-license. 
https://pkg.go.dev/license-policy mentions BSD-1-Clause, and the list of recognized license files (including COPYING).
Closing.
